src/WorkerPool.php

```diff
@@ -409,12 +443,43 @@ public function signalHandler($signo) {
 			case SIGUSR1:
 				// handle sigusr
 				break;
+			case SIGALRM:
+				$this->respawnIfRequired();
+				break;
 			default: // handle all other signals
 		}
 		// more signals to dispatch?
 		pcntl_signal_dispatch();
 	}
 
+        /**
+         * Respawn workers automatically if they died
+         *
+         * @param boolean $respawn
+         */
+        public function respawnAutomatically($respawn = true)
+        {
+            if ($this->respawnAutomatically = $respawn) {
+                $this->logger->info("WorkerPool configured to respawn");
+                pcntl_signal(SIGALRM, [$this, 'signalHandler']);
```

Comment:
rewrite this to pcntl_signal(SIGALRM, array($this, 'signalHandler')); to support older php versions and pass all tests